x-pack/plugins/apm/public/context/ChartsSyncContext.tsx

```diff
@@ -28,7 +28,7 @@ const ChartsSyncContextProvider: React.FC = ({ children }) => {
     callApmApi => {
       if (start && end && serviceName) {
         return callApmApi({
-          pathname: '/api/apm/services/{serviceName}/annotations',
+          pathname: '/api/apm/services/{serviceName}/annotation/search',
```

Comment:
I know this was not introduced in this PR but...
it seems wrong to fetch annotations data here. The responsibility of `ChartsSyncContext` was simply to keep the hover state in sync between multiple charts. 

What about pulling the annotations fetching out to a separate hook and call it where needed? I suspect it's only two places:

 - https://github.com/elastic/kibana/blob/7002fa878bf4829e423f209d9f6608c74a7ad907/x-pack/plugins/apm/public/components/shared/charts/TransactionCharts/TransactionLineChart/index.tsx#L45
 - https://github.com/elastic/kibana/blob/7002fa878bf4829e423f209d9f6608c74a7ad907/x-pack/plugins/apm/public/components/shared/charts/MetricsChart/index.tsx#L40

Reply:
I've thought about this for a bit but I'm not sold. To me, ChartsSyncContext is just a place where we store state common to all charts. I'm not sure if splitting this up would improve things, sounds easy to forget. Do you have any specific concerns, other than responsibility?